### PR TITLE
Fixes italic markdown to match newline and adds tests

### PR DESCRIPTION
**Pull request by alexxxwork**
@neil-marcellini @eVoloshchak please take a look.

Markdown symbols like `*text*`, `~text~` or `_text_` should match text with newlines.
Bold and strikethrough markdown have been fixed here https://github.com/Expensify/expensify-common/pull/486
this PR adds fix for italic text `_text_` and corresponding tests

### Fixed Issues
$ https://github.com/Expensify/App/issues/14675

# Tests
1. What unit/integration tests cover your change? What autoQA tests cover your change?
```
npm test -- __tests__/ExpensiMark-HTML-test.js
```
```
npm test -- __tests__/ExpensiMark-Markdown-test.js
```
1. What tests did you perform that validates your changed worked?
tested on native and web
<details>
<summary>ios</summary>
Screenshot 2023-02-15 06 05 23
</details>
<details>
<summary>ios web</summary>
Screenshot 2023-02-15 06 14 25
</details>
<details>
<summary>android</summary>
Screenshot 2023-02-15 06 33 40

</details>
<details>
<summary>android web</summary>
Screenshot 2023-02-15 06 29 41
</details>
<details>
<summary>web</summary>
Screenshot 2023-02-15 06 10 41
</details>

# QA
1. What does QA need to do to validate your changes?
Changes should be validated in App
1. What areas to they need to test for regressions?
Markdown italic in composer

### CODE REVIEW

**Comment by alexxxwork** [comment]
I have read the CLA Document and I hereby sign the CLA

File: lib/ExpensiMark.js [hunk 1]
```diff
@@ -237,9 +238,10 @@ export default class ExpensiMark {
                 regex: /\s*<(li)(?:"[^"]*"|'[^']*'|[^'">])*>(.*?)<\/\1>(?![^<]*(<\/pre>|<\/code>))\s*/gi,
                 replacement: '<li>  $2</li>',
             },
+            // Use [\s\S]* instead of .* to match newline
```
Comment (eVoloshchak) [inline lib/ExpensiMark.js:241]:
```suggestion
            
            // Use [\s\S]* instead of .* to match newline
```

**Review by eVoloshchak — CHANGES_REQUESTED** [review]

**Comment by eVoloshchak** [comment]
> I think as long as they comment on it you can assign them as a reviewer

✅

**Review by eVoloshchak — DISMISSED** [review]
Tests well!

**Review by neil-marcellini — APPROVED** [review]
Great! The code looks good and it tests well. Thank you.

**Review by neil-marcellini — CHANGES_REQUESTED** [review]
@alexxxwork Looks like you have a lint error. While you're at it, would you please move the multi-line italic tests directly below the single line italic test? That's how I've organized things so far.

**Comment by alexxxwork** [comment]
> @alexxxwork Looks like you have a lint error. While you're at it, would you please move the multi-line italic tests directly below the single line italic test? That's how I've organized things so far.

It's a bug in a fix :) Fixed!

**Comment by neil-marcellini** [comment]
> would you please move the multi-line italic tests directly below the single line italic test?

@alexxxwork

**Comment by alexxxwork** [comment]
> > would you please move the multi-line italic tests directly below the single line italic test?
> 
> @alexxxwork

Sorry, missed it! Just a moment

**Comment by alexxxwork** [comment]
@neil-marcellini All fixed!

**Comment by eVoloshchak** [comment]
There is still something weird with access to the repository, I can't see the lint check, only those 3
image

Local tests are looking good

**Review by eVoloshchak — DISMISSED** [review]

**Review by neil-marcellini — APPROVED** [review]
👍

**Comment by neil-marcellini** [comment]
Now it says that the commits need to be signed. I didn't think we required this for contributors so I'll ask in Slack.

**Comment by neil-marcellini** [comment]
@alexxxwork It looks like we do ask you to sign your commits in the contributing guidelines [here](https://github.com/Expensify/App/blob/1d83fe14bfcf0f7078ee4ae23ad9e89c50167677/contributingGuides/CONTRIBUTING.md?plain=1#L127-L129).

Once you have set up commit signing you can follow these instructions to sign the commits by rebasing and force pushing to this branch.

1. Check out a new branch to test on in case you mess up

2. Find the commit before the ones you want to change

3. `git rebase -i --preserve-merges <sha>`

4. Change `pick` to `edit` for the commits you want to sign

5. `git commit -S --amend --no-edit` when it prompts you

6. `git rebase --continue`

7. Push your new branch to github, confirm that it is fixed

8. Repeat steps 3-6 on your actual branch and force push.

**Comment by alexxxwork** [comment]
@neil-marcellini thank you very much for such a detailed instruction. It seems that I just commited from another PC without my gpg key. 
Meanwhile, found more 'automatic' way for such cases: `git rebase --exec 'git commit --amend --no-edit -n -S' -i <hash of the commit to rebase to>`

**Review by neil-marcellini — APPROVED** [review]
Good to go. Thanks for signing the commits and posting a quicker method! I'll post that to our internal documentation.

**Pull request merged**